### ISSUE TITLE
FTBFS: t/55utf8.t fails with MariaDB 10.3.12

### ISSUE THREAD

**Issue by gregoa** [gooddata/DBD-MariaDB#121]
We have the following bug reported to the Debian package of
DBD-MariaDB (https://bugs.debian.org/919071):

It doesn't seem to be a bug in the packaging, so you may want to take
a look. Thanks!

------8<-----------8<-----------8<-----------8<-----------8<-----

```
Source: libdbd-mariadb-perl
Version: 1.11-1
Severity: serious
Tags: upstream ftbfs
Justification: fails to build from source (but built successfully in the past)

-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

As noticed by Otto in #917303, the MariaDB update from 10.3.11 to
10.3.12 causes a test failure in libdbd-mariadb-perl:

t/55utf8.t ..............................
1..84
ok 1
ok 2
ok 3 - testing quoting of utf 8 string
ok 4 - testing quoting of blob
ok 5 - SET SQL_MODE=''
ok 6
ok 7
ok 8
ok 9
ok 10
ok 11
ok 12
ok 13
ok 14
ok 15 - got warning for INSERT
not ok 16 - warning is correct
ok 17
ok 18 - got data
ok 19
ok 20
ok 21
ok 22
ok 23
ok 24
ok 25
ok 26 - compare Ã<80>dam eq Ã<80>dam
ok 27
ok 28 - SET character_set_results='latin1'
ok 29 - got data
ok 30 - utf8 data are returned as latin1 when @@character_set_results is latin1
ok 31 - blob is unchanged when @@character_set_results is latin1
ok 32 - utf8 data are returned as latin1 when @@character_set_results is latin1
ok 33 - utf8 data are returned as latin1 when @@character_set_results is latin1
ok 34 - utf8 data are returned as latin1 when @@character_set_results is latin1
ok 35 - latin1 data are returned as latin1 when @@character_set_results is latin1
ok 36 - latin1 data are returned as latin1 when @@character_set_results is latin1
ok 37
ok 38
ok 39 -
ok 40 - SET character_set_results='utf8mb4'
ok 41
ok 42
ok 43
ok 44
ok 45 - testing quoting of utf 8 string
ok 46 - testing quoting of blob
ok 47 - SET SQL_MODE=''
ok 48
ok 49
ok 50
ok 51
ok 52
ok 53
ok 54
ok 55
ok 56
ok 57 - got warning for INSERT
not ok 58 - warning is correct
ok 59
ok 60 - got data
ok 61
ok 62
ok 63
ok 64
ok 65
ok 66
ok 67
ok 68 - compare Ã<80>dam eq Ã<80>dam
ok 69
ok 70 - SET character_set_results='latin1'
ok 71 - got data
ok 72 - utf8 data are returned as latin1 when @@character_set_results is latin1
ok 73 - blob is unchanged when @@character_set_results is latin1
ok 74 - utf8 data are returned as latin1 when @@character_set_results is latin1
ok 75 - utf8 data are returned as latin1 when @@character_set_results is latin1
ok 76 - utf8 data are returned as latin1 when @@character_set_results is latin1
ok 77 - latin1 data are returned as latin1 when @@character_set_results is latin1
ok 78 - latin1 data are returned as latin1 when @@character_set_results is latin1
ok 79
ok 80
ok 81 -
ok 82 - SET character_set_results='utf8mb4'
ok 83
ok 84
Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/84 subtests
[â¦]
Test Summary Report
- -------------------
t/55utf8.t                            (Wstat: 512 Tests: 84 Failed: 2)
  Failed tests:  16, 58
  Non-zero exit status: 2
Files=89, Tests=3262, 39 wallclock secs ( 0.64 usr  0.13 sys +  8.42 cusr  1.11 csys = 10.30 CPU)
Result: FAIL
Failed 1/89 test programs. 2/3262 subtests failed.



-----BEGIN PGP SIGNATURE-----

iQKTBAEBCgB9FiEE0eExbpOnYKgQTYX6uzpoAYZJqgYFAlw56IJfFIAAAAAALgAo
aXNzdWVyLWZwckBub3RhdGlvbnMub3BlbnBncC5maWZ0aGhvcnNlbWFuLm5ldEQx
RTEzMTZFOTNBNzYwQTgxMDREODVGQUJCM0E2ODAxODY0OUFBMDYACgkQuzpoAYZJ
qgZ8BxAAjTMWleXBRtmNVdeTo4L3qbr86thuRKMT0wauDoUY1fMPYgRSfziwLN7b
+7+8/MCb9ESOTGkacZulH3bPmRELV/LkDJN3XbAL1KcrQT0eMb2s4q2VXLybgoE8
q8pupqfwaEI2G7DGCOBralbMYtZ3UWyy2bJsQIBY+JcZk95yl1ulHoI+n1m//ETs
UpKlHrQPB01wGqDm+K95Cpxv5Zvym3a9n1Bzw2WCNIFf9NX85R+lKlKn7my+J9+j
Bllr80/BCcCUmjG/fIMxI2SiEmOgeTJwoLpQ14cpN1BlS3nWW+AXCHDX+G1MHGDS
9Lc4sh0b58MyBBfLnPvW+SUHhDX3R1ZZmKh3JLPH2tSKTeWDsEBBNih7D1hkB2hV
xiqqlQ0Y2SpUSVhvRUsKJ7Hdoqx4ko5X/9xG/kXC2ke6EYnKJzJ5JKwvogZ3AnDz
PSMh8EpcZTGZkD5Q1AiwHQaow/nmNEJJQXyov4qGeiwpX7ERt/7wWIbCCfHp9Zjp
jG8LEpgJu9A2iTxbevpFUEVVLw8i2dzmYnLW1XcpUdTTpJ7yqn5JYXGDzuNpvUyO
69X0A4o5m+3kObFdwQqgVfIMignbyQFRvDb5/eCuVz7zCiZsIl9bRt2YgsH3l4Kv
XcXnL7a5H3DUo3BsaHsPIvqkKXtn3Cn2zMpm7HRpbqA6ecm7VDI=
=L0uA
-----END PGP SIGNATURE-----

```
------8<-----------8<-----------8<-----------8<-----------8<-----

Thanks for considering,
  gregor herrmann,
  Debian Perl Group

**Comment by janchochol** [comment]
Thanks for report.
It seems, that this was caused by changed behavior in https://jira.mariadb.org/browse/MDEV-14576 .
We will need to extend tests to work also with this new error format.

**Comment by gregoa** [comment]
> It seems, that this was caused by changed behavior in https://jira.mariadb.org/browse/MDEV-14576 .
> We will need to extend tests to work also with this new error format.

Makes sense. -- Thanks for looking into this so quickly.

**Comment by pali** [comment]
Hi @gregoa! Fix for this test is in pull request: https://github.com/gooddata/DBD-MariaDB/pull/123. Can you verify that it fixes this problem? With MariaDB 10.3.12 on Travis all tests are passing: https://travis-ci.org/gooddata/DBD-MariaDB/jobs/480768047.

**Comment by gregoa** [comment]
On Thu, 17 Jan 2019 04:03:33 -0800, pali wrote:

> Hi @gregoa! Fix for this test is in pull request:
> https://github.com/gooddata/DBD-MariaDB/pull/123. Can you verify
> that it fixes this problem?

Thanks @pali.
Yes, the patch works in Debian/unstable as well.
